### DISCUSSION TITLE
Request: next/previous method with ]m [m ]M [M

### DISCUSSION THREAD

**Discussion by adam-c-anderson** [JetBrains/ideavim#834] [Ideas]
As described under [Various Motions](https://vimhelp.org/motion.txt.html#various-motions)

Navigation to start/end of next/previous method with ]m [m ]M [M would be supremely useful in many C-like languages including Java and C#

I just tried using them with IDEAVim 2.9.1 and they don't appear to have any effect, so I assume they are not implemented yet

**Comment by lippfi** [comment]
It is already implemented and should work. Are you a Rider user? There is an issue with Rider - https://youtrack.jetbrains.com/issue/VIM-2997/Previous-next-method-not-working-in-Rider

If you're not using Rider, could you please create a ticket with some additional details for me to reproduce and fix the issue? You can submit it here: https://youtrack.jetbrains.com/issues/VIM

**Reply by adam-c-anderson** [reply]
I am indeed using Rider! Thanks for the link to that issue. I'll use the workaround in the link for now and watch the issue for updates. Thanks!
